src/react-components/client-info-dialog.js

```diff
@@ -55,15 +56,18 @@ export default class ClientInfoDialog extends Component {
   }
 
   addOwner() {
-    const { user, performConditionalSignIn, hubChannel } = this.props;
+    const { user, performConditionalSignIn, hubChannel, onCloseDialog } = this.props;
     const { profile } = this.props.user;
 
     performConditionalSignIn(
       () => hubChannel.can("update_roles"),
       async () => {
-        this.props.showNonHistoriedDialog(PromoteClientDialog, {
+        this.props.showNonHistoriedDialog(PromoteClientModal, {
           displayName: profile.displayName,
-          onConfirm: () => hubChannel.addOwner(user.id)
+          onConfirm: () => {
+            hubChannel.addOwner(user.id);
+            onCloseDialog();
```

Comment:
The PeopleSidebar may need a re-render if it's open. (Maybe not a direct call to re-render here, but instead could be downstream of updates from hub-channel.)

Reply:
I forgot that setting a selected person would mean that it would store a stale object when that person is updated. Now I just store the id of the person.